src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[16];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = generateIndex(key);
+        if (buckets[index] != null) {
```

Comment:
```suggestion
        if (buckets[index] == null) {
        	buckets[index] = new Node<>(Objects.hash(key), key, value, null);
        	return;
        }
        ....
```

It is really hard to read if-else constructions, better to avoid them if we can